packages/blade/src/components/Tooltip/_decisions/decisions.md

```diff
@@ -7,63 +7,66 @@ The tooltip typically provides additional context about the element or its funct
 
 ## `Tooltip` API
 
-| Prop          | Type                                                                     | Default     | Description                                                               | Required |
-| ------------- | ------------------------------------------------------------------------ | ----------- | ------------------------------------------------------------------------- | -------- |
-| content       | `string`                                                                 | `undefined` | Content of the tooltip                                                    | ✅       |
-| placement     | `top, top-start, top-end, left, right, bottom, bottom-start, bottom-end` | `top`       | Placement of tooltip                                                      | ✅       |
-| children      | `React.ReactNode`                                                        | `undefined` | Trigger component for tooltip, Accepts any interactive element            | ✅       |
-| isOpen        | `boolean`                                                                | `undefined` | Controls the tooltip state                                                |          |
-| defaultIsOpen | `boolean`                                                                | `undefined` | If true, the tooltip will be visible initially                            |          |
-| onDismiss     | `Callback`                                                               | `undefined` | Called when tooltip is closed, either by mouseout or users pressing `Esc` |          |
+| Prop      | Type                                                                     | Default     | Description                                                             | Required |
+| --------- | ------------------------------------------------------------------------ | ----------- | ----------------------------------------------------------------------- | -------- |
+| content   | `string`                                                                 | `undefined` | Content of the tooltip                                                  | ✅       |
+| placement | `top, top-start, top-end, left, right, bottom, bottom-start, bottom-end` | `top`       | Placement of tooltip                                                    | ✅       |
+| children  | `React.ReactNode`                                                        | `undefined` | Trigger component for tooltip, Accepts any interactive element or icons | ✅       |
+| onOpen    | `Callback`                                                               | `undefined` | Called when tooltip is opened                                           |          |
+| onClose   | `Callback`                                                               | `undefined` | Called when tooltip is closed                                           |          |
```

Comment:
Something to ponder: should we have a single `onChange` callback with `{ isOpen / isVisible}` callback param over this and why?

Reply:
Hmm yeah.. Kind of leaning towards two callbacks. 

I feel it's a matter of API choice, kind of subjective but here's my thought. 

**1. Naming**

onChange naming doesn't go very well with Tooltip, when we say on"Change" what exactly is changing? `onStateChange` might be better? 
<Tooltip onChange={(isOpen) => console.log(isOpen)} \/> 

**2. Ergonomics**

I feel (and this might be subjective) two callbacks provide a better ergonomic API that 1. 

```jsx
<Tooltip 
  onOpen={sendAnalyticsOnOpen} 
  onClose={sendAnalyticsOnClose} 
 />
```

vs

```jsx
<Tooltip 
  onChange={(isOpen) => {
    if (isOpen) {
      sendAnalyticsOnOpen()
    } else {
      sendAnalyticsOnClose();
    }
  }} 
 />
```

Reply:
Btw on collapsible I've proposed `onChange` rather than different callbacks for expanding / collapsing, let's also gather more feedback from the team

Reply:
@chaitanyadeorukhkar @saurabhdaware any thoughts?